src/store/modules/emails.js

```diff
@@ -15,13 +26,70 @@ const state = [
 ];
 
 // getters
-const getters = {};
+const getters = {
+  getSenderObject(state) {
+    //returns first object, which will be the selected sender
+    return state[0];
+  },
+  getEmailsMostRecentFirst() {
+    //sorts the
+    const sortedMail = state[0].emails.sort((a, b) => {
+      if (a.receivedTime > b.receivedTime) {
+        return -1;
+      }
+      if (a.receivedTime < b.receivedTime) {
+        return 1;
+      }
+      return 0;
+    });
+
+    return sortedMail;
+  },
+};
 
 // actions
-const actions = {};
+const actions = {
+  //these functions would probably be used by the email service
+  sendEmail({ commit }, payload) {
+    //adds email to display, nothing else really happens here
+    commit("addEmail", payload);
+  },
+  senderIsSelected({ commit }, selectedSenderId) {
+    commit("moveSenderToFirst", selectedSenderId);
+  },
+};
 
 // mutations
-const mutations = {};
+const mutations = {
+  addEmail(state, payload) {
+    //gets whoever the sender is, based on payload's id
+    const sender = state.find((sender) => payload.senderId === sender.senderId);
+
+    //finds the largest emailId amongst the ones that already exist, adds one to it
+    const emailId =
+      Math.max(...sender.messages.map((email) => email.emailId)) + 1;
+
+    const newEmail = {
+      emailId: emailId,
+      messageTitle: payload.messageTitle,
+      messageBody: payload.messageBody,
+      receivedTime: Date.now(),
+    };
+
+    sender.push(newEmail);
+  },
+  moveSenderToFirst(state, selectedSenderId) {
```

Comment:
Using a getter is an easier way to select and item instead of moving sender reordering the state.
This can be done by searching for an email id based off of Inbox/selectedInboxItemId in your getEmailsMostRecentFirst getter. If you have questions please ask